mongorestore/mongorestore_test.go

```diff
@@ -1084,3 +1087,65 @@ func TestSkipSystemCollections(t *testing.T) {
 		})
 	})
 }
+
+// TestSkipStartAndAbortIndexBuild asserts that all "startIndexBuild" and "abortIndexBuild" oplog
+// entries are skipped when restoring the oplog.
+func TestSkipStartAndAbortIndexBuild(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	ctx := context.Background()
+
+	sessionProvider, _, err := testutil.GetBareSessionProvider()
+	if err != nil {
+		t.Fatalf("No cluster available: %v", err)
+	}
+	session, err := sessionProvider.GetSession()
+	if err != nil {
+		t.Fatalf("No client available")
+	}
+
+	if ok, _ := sessionProvider.IsReplicaSet(); !ok {
+		t.SkipNow()
+	}
+
+	Convey("With a test MongoRestore instance", t, func() {
+		testdb := session.Database("test")
+
+		// Drop the collection to clean up resources
+		defer testdb.Collection("skip_index_entries").Drop(ctx)
+
+		// oplog.bson only has startIndexBuild and abortIndexBuild entries
+		args := []string{
+			DirectoryOption, "testdata/oplog_ignore_index",
+			OplogReplayOption,
+			DropOption,
+		}
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		if restore.serverVersion.GTE(db.Version{4, 4, 0}) {
+			// Run mongorestore
+			dbLocal := session.Database("local")
+			queryObj := bson.D{{
+				"$or", bson.A{
+					bson.D{{"ns", primitive.Regex{Pattern: "^config.system.sessions*"}}},
```

Comment:
But this is including config.system.sessions, not excluding it. Currently this query is matching `ns == /^config.system.sessions*/ || op != "n"`. I think we need `ns != "config.system.sessions" && op != "n"`

Reply:
Oh, my mistake! I guess I didn't notice the error since there were no `config.system.sessions` logs in the oplog when testing 😅 I think the two conditions should be `or`'d, though.